more_itertools/more.py

```diff
@@ -1591,18 +1591,43 @@ def sort_together(iterables, key_list=(0,), reverse=False):
         >>> sort_together(iterables, key_list=(1, 2))
         [(2, 3, 1), (0, 0, 1), ('a', 'c', 'b')]
 
+    To sort by a function of the elements of the iterable, pass a *key*
+    function. Its arguments are the elements of the iterables corresponding to
+    the key list::
+
+        >>> iterables = [(3, 1, 2), (1, 0, 2), ('c', 'b', 'a')]
+        >>> sort_together(iterables, key_list=(0, 1), key=lambda x, y: x - y)
```

Comment:
This example is a little confusing. What's being subtracted in the last item?

Reply:
The lists are being sorted by the difference between the value in the first list and the second list at each index (since the key list is 0 and 1, x and y are elements of the first and second lists respectively).

It would raise a TypeError on that input, because it would be subtracting a number from a string.

Reply:
Could you put a different example in the docstring? This one is hard (for me) to reason about - it's sort of a fluke that not all the items get touched by the key function?

Reply:
Yeah sure, I'll find a better example.

Can you explain what you mean by "not all the items get touched by the key function"? You mean that the key function gets only two arguments? That's because of the key list.

Maybe I'll add a second example to demonstrate the interaction of the key list and the key function? Or maybe it would be clearer if they didn't interact, and the key function always got all of the elements as it's arguments.

Reply:
I made the example clearer, let me know what you think